internal/bminventory/inventory.go

```diff
@@ -4687,3 +4711,237 @@ func (b *bareMetalInventory) AddOpenshiftVersion(ctx context.Context, ocpRelease
+			host.Kind = swag.String(models.HostKindAddToExistingClusterHost)
+		}
+		host.ClusterID = infraEnv.ClusterID
+	}
+
+	if err = b.hostApi.RegisterHost(ctx, host, tx); err != nil {
+		log.WithError(err).Errorf("failed to register host <%s> infra-env <%s>",
+			params.NewHostParams.HostID.String(), params.InfraEnvID.String())
+		uerr := errors.Wrap(err, fmt.Sprintf("Failed to register host %s", hostutil.GetHostnameForMsg(host)))
+
+		// TODO event for infra-env
+		b.eventsHandler.AddEvent(ctx, params.InfraEnvID, params.NewHostParams.HostID, models.EventSeverityError,
+			uerr.Error(), time.Now())
+		return returnRegisterHostTransitionError(http.StatusBadRequest, err)
+	}
+
+	if err = b.customizeHost(host); err != nil {
+		b.eventsHandler.AddEvent(ctx, params.InfraEnvID, params.NewHostParams.HostID, models.EventSeverityError,
+			"Failed to register host: error setting host properties", time.Now())
+		return common.GenerateErrorResponder(err)
+	}
+
+	b.eventsHandler.AddEvent(ctx, params.InfraEnvID, params.NewHostParams.HostID, models.EventSeverityInfo,
+		fmt.Sprintf("Host %s: registered to cluster", hostutil.GetHostnameForMsg(host)), time.Now())
+
+	hostRegistration := models.HostRegistrationResponse{
+		Host:                  *host,
+		NextStepRunnerCommand: b.generateV2NextStepRunnerCommand(ctx, &params),
+	}
+
+	if err := tx.Commit().Error; err != nil {
+		log.Error(err)
+		return installer.NewRegisterHostInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+
+	if infraEnv.ClusterID != "" {
+		if err := b.crdUtils.CreateAgentCR(ctx, log, params.NewHostParams.HostID.String(), cluster.KubeKeyNamespace, cluster.KubeKeyName, cluster.ID); err != nil {
+			log.WithError(err).Errorf("Fail to create Agent CR. Namespace: %s, Cluster: %s, HostID: %s", cluster.KubeKeyNamespace, cluster.KubeKeyName, params.NewHostParams.HostID.String())
+			return installer.NewRegisterHostInternalServerError().
+				WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+		}
+	}
+
+	txSuccess = true
+
+	return installer.NewV2RegisterHostCreated().WithPayload(&hostRegistration)
+}
+
+func (b *bareMetalInventory) V2GetNextSteps(ctx context.Context, params installer.V2GetNextStepsParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	var steps models.Steps
+
+	txSuccess := false
+	tx := b.db.Begin()
+	defer func() {
+		if !txSuccess {
+			log.Error("get next steps failed")
+			tx.Rollback()
+		}
+		if r := recover(); r != nil {
+			log.Error("get next steps failed")
+			tx.Rollback()
+		}
+	}()
+
+	if tx.Error != nil {
+		log.WithError(tx.Error).Errorf("failed to start db transaction")
+		return installer.NewUpdateClusterInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, errors.New("DB error, failed to start transaction")))
+	}
+
+	//TODO check the error type
+	host, err := common.GetV2HostFromDB(tx, params.InfraEnvID.String(), params.HostID.String())
```

Comment:
this function should be called GetHostFromDB

Reply:
Whatever we call it, there is another function with the same name that extracts host according to cluster-id

Reply:
There are still many locations in the code that use it as cluster id.  It is not correct to change the implementation without changing the code.

Reply:
this is a core function, there is should be only one such. since currently in all the hosts have cluster-id=infra-env-id, we can use the same function with passing it infra-env-id or cluster-id. the main thing is that it is the same function. MGMT-7321 will take care of moving all the code ( where relevant ) to pass infra-en-id instead of cluster-id